endpoints/v2/proxy.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+from app import app
+from data.database import ProxyCacheConfig
+from data.model import InvalidOrganizationException
+from data.model.proxy_cache import get_proxy_cache_config_for_org
+from proxy import Proxy
+
+
+class ProxyNotSupported(Exception):
+    pass
+
+
+def _get_proxy_config(org_name: str) -> ProxyCacheConfig:
+    if not app.config.get("FEATURE_PROXY_CACHE", False):
+        raise ProxyNotSupported("Proxy globally disabled.")
+
+    try:
+        config = get_proxy_cache_config_for_org(org_name)
+        return config
+    except (InvalidOrganizationException, ProxyCacheConfig.DoesNotExist):
+        # we may end up here if `org_name` either refers to a user
+        # or if the org is not set up as cache - either way proxy shouldn't work.
+        raise ProxyNotSupported("Organization is invalid or not configured as proxy cache.")
```

Comment:
Do you think we can propagate the same error that was raised instead of raising a new one? Since we are catching `InvalidOrganizationException` and `ProxyCacheConfig.DoesNotExist` we propagate it to be more specific.

Reply:
There are multiple reasons for a `ProxyNotSupported` exception to happen:
 * the `FEATURE_PROXY_CACHE` is set to False;
 * `org_name` was a user rather than an org;
 * there is no proxy cache config for the org;
By not returning `ProxyNotSupported` exception here, the caller would have to handle 3 different exceptions to cover all cases of where proxy is not supported. Returning a single exception for all 3 situations is the way I found to make it easier for the caller to know whether proxy code needs to be executed or not, regardless of reason.

I'm open to new ideas, so long as we easily distinguish cases of when the everything looks good and the proxy code should be executed vs when the "normal" flow should take over instead.

What do you think?